### PR TITLE
Add sample for downloading and managing languages with the Android Keyboard SDK.

### PR DESCRIPTION
**Pull request by iagotorrado-fleksy**
feature: Add sample for downloading and managing languages with the Android Keyboard SDK.

### CODE REVIEW

File: Examples/Languages/android/DownloadLanguage/app/src/androidTest/java/com/fleksy/samples/downloadlanguage/ExampleInstrumentedTest.kt
Comment (francescojalvo) [inline Examples/Languages/android/DownloadLanguage/app/src/androidTest/java/com/fleksy/samples/downloadlanguage/ExampleInstrumentedTest.kt:1]: Is it good practice to keep this boilerplate code?

**Review by francescojalvo — APPROVED** [review]

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    fun addLanguageToKeyboard(locale: String, layout: String? = null) {
+        LanguagesHelper.addLanguage(KeyboardLanguage(locale, layout))
+    }
+
+    /**
+     * Removes the downloaded language from the LanguagesConfiguration.userLanguages and deletes
+     * the local files.
+     *
+     * Note: After downloading a language, if the SDK's configuration has not been updated,
+     * this method will be unable to delete said language.
```
Comment (ajpallares-fleksy) [inline Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt:115]: I think this phrase is not clear.

> Note: After downloading a language, if the SDK's configuration has not been updated, this method will be unable to delete said language.

I guess you mean that the language won't be deleted if said language is not included in the `userLanguages` parameter of the configuration. I think that saying "if the SDK's configuration has not been updated". Perhaps adding "if the SDK's configuration has not been updated to include said language", something like

> Note: After downloading a language, if the SDK's configuration has not been updated to include said language, this method will be unable to delete it.

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    fun getAvailableLayoutsForLanguage(locale: String, callback: (List<String>?) -> Unit) {
+        LanguagesHelper.languageResourceDetails(locale) {
+            callback.invoke(it?.layouts)
+        }
+    }
+
+    /**
+     * Changes the keyboard's current language and/or it's layout.
+     */
+
```
Comment (ajpallares-fleksy) [inline Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt:134]: Extra newline to be removed

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        LanguagesHelper.changeLanguage(KeyboardLanguage(locale, layout))
+    }
+
+    /**
+     * Change the layout for any locally available language. When changing the layout for the
+     * keyboard's current language, the keyboard's UI will be refreshed.
+     *
+     * Note: After downloading a language, if the SDK's configuration has not been updated,
+     * this method will be unable to change the layout for said language.
+     */
```
Comment (ajpallares-fleksy) [inline Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt:145]: The same comment as above

> Note: After downloading a language, if the SDK's configuration has not been updated to include said language, this method will be unable to change the layout for it.

File: Examples/Languages/android/DownloadLanguage/app/src/test/java/com/fleksy/samples/downloadlanguage/ExampleUnitTest.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
+    @Test
+    fun addition_isCorrect() {
+        assertEquals(4, 2 + 2)
+    }
+}
```
Comment (ajpallares-fleksy) [inline Examples/Languages/android/DownloadLanguage/app/src/test/java/com/fleksy/samples/downloadlanguage/ExampleUnitTest.kt:17]: I don't think we need this

**Review by xevifar — APPROVED** [review]

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                                "Error downloading language: $locale",
+                                Toast.LENGTH_LONG
+                            ).show()
+                            Log.e(
+                                "LanguagesManager",
+                                "Error downloading language: $locale. Error: ${error.localizedMessage}"
+                            )
+                        }
+                    })
+                }
```
[the remark this answers is not in the document]
Comment (iagotorrado-fleksy) [reply Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt:84]: A toast is already being shown in the language download onError. Where exactly do you mean?

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/SdkKeyboardService.kt [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                licenseKey = "INSERT_LICENSE_KEY",
+                licenseSecret = "INSERT_LICENSE_SECRET"
+            ),
+            style = KeyboardConfiguration.StyleConfiguration(
+                swipeDuration = 300
+            ),
+            language = KeyboardConfiguration.LanguageConfiguration(
+                current = currentLanguage,
+                userLanguages = userLanguages,
+                automaticDownload = false,
```
[the remark this answers is not in the document]
Comment (iagotorrado-fleksy) [reply Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/SdkKeyboardService.kt:63]: Added a comment explainig it.

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (iagotorrado-fleksy) [reply Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt:84]: Saw it. Thank you!

File: Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt (hunk 1 above)
Comment (iagotorrado-fleksy) [reply Examples/Languages/android/DownloadLanguage/app/src/main/java/com/fleksy/samples/downloadlanguage/languages/LanguagesManager.kt]: Updated comment.

**Review by ajpallares-fleksy — APPROVED** [review]

**Pull request merged**